R/internals.R

```diff
@@ -411,50 +412,15 @@
   xml2::read_html(session)
 }
 
-# Use 1.8.0 version of jsonlite::fromJSON since 1.8.2's version (that uses base::url()) doesn't work for some cases
-#' @importFrom jsonlite validate parse_json
-#' @importFrom curl new_handle handle_setheaders curl
-.fromJSON <- function(txt, simplifyVector = TRUE, simplifyDataFrame = simplifyVector, simplifyMatrix = simplifyVector, flatten = FALSE, ...) {
-
-  # check type
-  if (!is.character(txt) && !inherits(txt, "connection")) {
-    stop("Argument 'txt' must be a JSON string, URL or file.")
-  }
-
-  # overload for URL or path
-  if (is.character(txt) && length(txt) == 1 && nchar(txt, type="bytes") < 2084 && !jsonlite::validate(txt)) {
-    if (grepl("^https?://", txt, useBytes=TRUE)) {
-      agent <- getOption("worldfootballR.agent", default = "RStudio Desktop (2022.7.1.554); R (4.1.1 x86_64-w64-mingw32 x86_64 mingw32)")
-      h <- curl::new_handle(useragent = agent)
-      curl::handle_setheaders(h, Accept = "application/json, text/*, */*")
-      txt <- curl::curl(txt, handle = h)
-    } else if (file.exists(txt)) {
-      # With files we can never know for sure the encoding. Lets try UTF8 first.
-      # txt <- raw_to_json(readBin(txt, raw(), file.info(txt)$size));
-      txt <- file(txt)
-    }
-  }
-
-  jsonlite::parse_json(
-    txt = txt,
-    flatten = flatten,
-    simplifyVector = simplifyVector,
-    simplifyDataFrame = simplifyDataFrame,
-    simplifyMatrix = simplifyMatrix,
-    ...
-  )
-}
-
-# Sometimes .fromJSON doesn't work, but jsonlite::fromJSON will. See https://github.com/JaseZiv/worldfootballR/issues/201
-#' @importFrom purrr safely
+#' @importFrom httr GET content
 #' @importFrom jsonlite fromJSON
-#'
 #' @noRd
-safely_from_json <- function(...) {
-  f <- purrr::safely(.fromJSON, otherwise = NULL, quiet = TRUE)
-  resp <- f(...)
-  if (!is.null(resp)) {
-    return(resp)
-  }
-  jsonlite::fromJSON(...)
+get_content <- function(url) {
```

Comment:
this is the big change in this PR. this is basically our drop-in replacement for `jsonlite::fromJSON()`